c/omni_lock_time_lock.h

```diff
@@ -1,73 +1,18 @@
 #ifndef OMNI_LOCK_TIME_LOCK_H_
 #define OMNI_LOCK_TIME_LOCK_H_
 
-#define ERROR_INCORRECT_SINCE_FLAGS -23
-#define ERROR_INCORRECT_SINCE_VALUE -24
-/* since */
-#define SINCE_VALUE_BITS 56
-#define SINCE_VALUE_MASK 0x00ffffffffffffff
-#define SINCE_EPOCH_FRACTION_FLAG 0b00100000
-
-/* a and b are since value,
- return 0 if a is equals to b,
- return -1 if a is less than b,
- return 1 if a is greater than b */
-int epoch_number_with_fraction_cmp(uint64_t a, uint64_t b) {
-  static const size_t NUMBER_OFFSET = 0;
-  static const size_t NUMBER_BITS = 24;
-  static const uint64_t NUMBER_MAXIMUM_VALUE = (1 << NUMBER_BITS);
-  static const uint64_t NUMBER_MASK = (NUMBER_MAXIMUM_VALUE - 1);
-  static const size_t INDEX_OFFSET = NUMBER_BITS;
-  static const size_t INDEX_BITS = 16;
-  static const uint64_t INDEX_MAXIMUM_VALUE = (1 << INDEX_BITS);
-  static const uint64_t INDEX_MASK = (INDEX_MAXIMUM_VALUE - 1);
-  static const size_t LENGTH_OFFSET = NUMBER_BITS + INDEX_BITS;
-  static const size_t LENGTH_BITS = 16;
-  static const uint64_t LENGTH_MAXIMUM_VALUE = (1 << LENGTH_BITS);
-  static const uint64_t LENGTH_MASK = (LENGTH_MAXIMUM_VALUE - 1);
-
-  /* extract a epoch */
-  uint64_t a_epoch = (a >> NUMBER_OFFSET) & NUMBER_MASK;
-  uint64_t a_index = (a >> INDEX_OFFSET) & INDEX_MASK;
-  uint64_t a_len = (a >> LENGTH_OFFSET) & LENGTH_MASK;
-
-  /* extract b epoch */
-  uint64_t b_epoch = (b >> NUMBER_OFFSET) & NUMBER_MASK;
-  uint64_t b_index = (b >> INDEX_OFFSET) & INDEX_MASK;
-  uint64_t b_len = (b >> LENGTH_OFFSET) & LENGTH_MASK;
-
-  if (a_epoch < b_epoch) {
-    return -1;
-  } else if (a_epoch > b_epoch) {
-    return 1;
-  } else {
-    /* a and b is in the same epoch,
-       compare a_index / a_len <=> b_index / b_len
-     */
-    uint64_t a_block = a_index * b_len;
-    uint64_t b_block = b_index * a_len;
-    /* compare block */
-    if (a_block < b_block) {
-      return -1;
-    } else if (a_block > b_block) {
-      return 1;
-    } else {
-      return 0;
-    }
-  }
-}
+#include "ckb_utils.h"
+#define ERROR_INCORRECT_SINCE_FLAGS (-23)
+#define ERROR_INCORRECT_SINCE_VALUE (-24)
 
 /* check since,
  for all inputs the since field must have the exactly same flags with the since
  constraint, and the value of since must greater or equals than the since
- contstaint */
+ constraint */
 int check_since(uint64_t since) {
```

Comment:
I found the function hard to reason. Maybe renaming the paramemter can make it easier.

```suggestion
int check_since(uint64_t minimal_input_since_constraint) {
```

Reply:
I'm feeling not good about a so long  local variable. Can we have it shorter like "input_since"?

Reply:
How about `min_since`